silhouette/app/com/mohiva/play/silhouette/api/services/AuthenticatorService.scala

```diff
@@ -20,9 +20,66 @@
 package com.mohiva.play.silhouette.api.services
 
 import com.mohiva.play.silhouette.api.{ Authenticator, LoginInfo }
-import play.api.mvc.{ Headers, RequestHeader, Result }
+import play.api.libs.concurrent.Execution.Implicits._
+import play.api.libs.iteratee.Enumerator
+import play.api.mvc._
 
 import scala.concurrent.Future
+import scala.language.implicitConversions
+import scala.util.Try
+
+/**
+ * A marker result which indicates that an operation on an authenticator was processed and
+ * therefore it shouldn't updated automatically.
+ *
+ * Due the fact that the update method gets called on every subsequent request to update the
+ * authenticator related data in the backing store and in the result, it isn't possible to
+ * discard or renew the authenticator simultaneously. This is because the "update" method would
+ * override the result created by the "renew" or "discard" method, because it will be executed
+ * as last in the chain.
+ *
+ * As example:
+ * If we discard the session in a Silhouette action then it will be removed from session. But
+ * at the end the update method will embed the session again, because it gets called with the
+ * result of the action.
+ *
+ * @param result The result to wrap.
+ */
+class AuthenticatorResult(result: Result) extends Result(result.header, result.body, result.connection) {
+
+  /**
+   * Creates a new copy of a `AuthenticatorResult`.
+   *
+   * @param header The response header, which contains status code and HTTP headers.
+   * @param body The response body.
+   * @param connection The connection semantics to use.
+   * @return A copy of a `AuthenticatorResult`.
+   */
+  override def copy(header: ResponseHeader, body: Enumerator[Array[Byte]], connection: HttpConnection.Connection) = {
```

Comment:
I think this should have default params:
override def copy(header: ResponseHeader = this.header, body: Enumerator[Array[Byte]]=this.body, connection: HttpConnection.Connection=this.connection)